packages/uxpin-merge-cli/src/utils/fs/PresetFileGenerator/PresetFileGenerator.ts

```diff
@@ -0,0 +1,148 @@
+import { kebabCase } from 'lodash';
+import { dirname, relative, resolve } from 'path';
+import { ComponentImplementationInfo } from '../../../steps/discovery/component/ComponentInfo';
+import { getImplementationInfo } from '../../../steps/discovery/component/implementation/getImplementationInfo';
+import { ComponentMetadata } from '../../../steps/serialization/component/ComponentDefinition';
+import { getComponentMetadata } from '../../../steps/serialization/component/implementation/getComponentMetadata';
+import { printLine, printWarning } from '../../console/printLine';
+import { PrintColor } from '../../console/PrintOptions';
+import { writeToFile } from '../writeToFile';
+import { generatePropertyValuePair, INDENTATION_CHAR } from './helpers/generatePropertyValuePair';
+import { getPropTypeValue } from './helpers/getPropTypeValue';
+
+const PRESET_FILE_NAME:string = '0-default.jsx';
+
+interface Property {
+  name:string;
+  value:any;
+}
+
+function isChildrenProp(property:Property):boolean {
+  return property.name === 'children';
+}
+
+export class PresetFileGenerator {
+
+  private componentDirectory:string;
+  private componentPath:string;
+  private componentName!:string;
+  private componentMetadata!:ComponentMetadata;
+  private componentProperties!:Property[];
+  private presetDirectory:string;
+  private presetFilePath:string;
+
+  constructor(componentPath:string) {
+    this.componentPath = componentPath;
+    this.componentDirectory = dirname(componentPath);
+    this.presetDirectory = `${this.componentDirectory}/presets`;
+    this.presetFilePath = `${this.presetDirectory}/${PRESET_FILE_NAME}`;
+  }
+
+  public async init():Promise<void> {
+    if (await pathExists(resolve(__dirname, this.componentPath))) {
+      throw new Error(`Component ${this.componentPath} does not exists`);
+    }
+
+    const implementationInfo:ComponentImplementationInfo | null = getImplementationInfo(this.componentPath);
+
+    if (!implementationInfo) {
+      throw Error(`Invalid component path - ${this.componentPath}`);
+    }
+
+    const { result: metadata, warnings: metadataWarnings } = await getComponentMetadata(implementationInfo);
+
+    if (metadataWarnings.length && !metadata.properties.length) {
+      throw new Error(metadataWarnings[0].message);
+    }
+
+    this.componentMetadata = metadata;
+    this.componentName = this.componentMetadata.name;
+    this.setPropertiesValues();
+  }
+
+  public async createPresetFile():Promise<void> {
+    if (!await pathExists(this.presetDirectory)) {
+      await mkdir(this.presetDirectory);
+    }
+
+    if (await pathExists(this.presetFilePath)) {
+      printWarning(`File ${this.presetFilePath} exists`);
+      return;
+    }
+
+    const componentFileContent:string = await this.generateComponentFile();
+    await writeToFile(this.presetFilePath, componentFileContent);
+    printLine(`File ${this.presetFilePath} created successfully`, { color: PrintColor.GREEN });
```

Comment:
@robhil 

Another very minor thing, but for consistency with other messages that the CLI prints, could you utilise the green tick box ✅ for successful messages like this one, and the warning triangle 🛑 or info sign 🛈 for messages that say that the file already exists. You can search the project for examples. Thanks!

Reply:
I saw that for warnings we used 👉 so I used it as well